tests/e2e/scale/noobaa/test_scale_obc_creation_repsin_noobaa_pods.py

```diff
@@ -0,0 +1,118 @@
+import pytest
+import time
+
+from ocs_ci.ocs import constants, scale_noobaa_lib
+from ocs_ci.framework.testlib import scale, E2ETest
+from ocs_ci.ocs.resources.objectconfigfile import ObjectConfFile
+from ocs_ci.framework.pytest_customization.marks import vsphere_platform_required
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(autouse=True)
+def teardown(request):
+    def finalizer():
+        scale_noobaa_lib.cleanup(constants.OPENSHIFT_STORAGE_NAMESPACE)
+
+    request.addfinalizer(finalizer)
+
+
+@scale
+class TestScaleOCBCreation(E2ETest):
+    """
+    OBC scale creation, creating up to max support number of OBCs.
+    OBCs are created in the Multicloud Object Gateway and
+    Ceph Object Gateway (RGW)
+
+    """
+
+    namespace = constants.OPENSHIFT_STORAGE_NAMESPACE
+    scale_obc_count = 1000
+    num_obc_batch = 50
+
+    @pytest.mark.parametrize(
+        argnames=["pod_name", "sc_name"],
+        argvalues=[
+            pytest.param(
+                *["noobaa-core", constants.NOOBAA_SC],
+                marks=[
+                    pytest.mark.polarion_id("OCS-2645"),
+                ],
+            ),
+            pytest.param(
+                *["noobaa-db", constants.NOOBAA_SC],
+                marks=[
+                    pytest.mark.polarion_id("OCS-2646"),
+                ],
+            ),
+            pytest.param(
+                *["noobaa-core", constants.DEFAULT_STORAGECLASS_RGW],
+                marks=[
+                    vsphere_platform_required,
+                    pytest.mark.polarion_id("OCS-2647"),
+                ],
+            ),
+            pytest.param(
+                *["noobaa-db", constants.DEFAULT_STORAGECLASS_RGW],
+                marks=[
+                    vsphere_platform_required,
+                    pytest.mark.polarion_id("OCS-2648"),
+                ],
+            ),
+        ],
+    )
+    def test_scale_rgw_obc_creation(self, tmp_path, pod_name, sc_name, timeout=60):
+        """
+        OBC creation using RGW storage class
+        This test case only runs on vSphere cluster deployment
+        """
+
+        log.info(
+            f"Start creating  {self.scale_obc_count} "
+            f"OBC in a batch of {self.num_obc_batch}"
+        )
+        for i in range(int(self.scale_obc_count / self.num_obc_batch)):
+            obc_dict_list = (
+                scale_noobaa_lib.construct_obc_creation_yaml_bulk_for_kube_job(
+                    no_of_obc=self.num_obc_batch,
+                    sc_name=sc_name,
+                    namespace=self.namespace,
+                )
+            )
+            # Create job profile
+            job_file = ObjectConfFile(
+                name="job_profile",
+                obj_dict_list=obc_dict_list,
+                project=self.namespace,
+                tmp_path=tmp_path,
+            )
+            # Create kube_job
+            job_file.create(namespace=self.namespace)
+            time.sleep(timeout * 5)
```

Comment:
Do we need this sleep ? I feel better to handle it part of OBC bound validation.
```
def check_all_obc_reached_bound_state_in_kube_job(
    kube_job_obj, namespace, no_of_obc, timeout=120, no_wait_time=20
):
```

Reply:
Removed the sleep time in test_scale_obc_creation_repsin_noobaa_pods.py, changed the logic to handle waiting time in check_all_obc_reached_bound_state_in_kube_job.